modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/utils/bean/APICreationRequestBean.java

```diff
@@ -0,0 +1,342 @@
+    private String tiersCollection = "Gold";
+    private String resourceCount = "0";
+    private String roles = "";
+    private String wsdl = "";
+    private String default_version = "";
+    private String default_version_checked = "";
+    private List<APIResourceBean> resourceBeanList;
+    private String epUsername = "";
+    private String epPassword = "";
+    private String sandbox = "";
+    private String provider = "admin";
+
+
+    /**
+     * constructor of the APICreationRequestBean class
+     *
+     * @param apiName     - Name of the APi
+     * @param context     - API context
+     * @param version     - API version
+     * @param endpointUrl - Endpoint URL of the API
+     * @throws APIManagerIntegrationTestException
+     */
+    public APICreationRequestBean(String apiName, String context, String version, URL endpointUrl) throws APIManagerIntegrationTestException {
+        this.name = apiName;
+        this.context = context;
+        this.version = version;
+        resourceBeanList = new ArrayList<APIResourceBean>();
+        resourceBeanList.add(new APIResourceBean("GET", "Application & Application User", "Unlimited", "/*"));
+        try {
+            this.endpoint = new JSONObject("{\"production_endpoints\":{\"url\":\""
+                    + endpointUrl + "\",\"config\":null},\"endpoint_type\":\""
+                    + endpointUrl.getProtocol() + "\"}");
+        } catch (JSONException e) {
+            log.error("JSON construct error", e);
+            throw new APIManagerIntegrationTestException(" Error When constructing the edn point url JSON", e);
+        }
+
+
+    }
+
+    @Override
+    /**
+     * Set the Action as addAPI
+     */
+    public void setAction() {
+        setAction("addAPI");
+    }
+
+    /**
+     * Set a custom Action
+     *
+     * @param action - custom action
+     */
+    public void setAction(String action) {
+        super.setAction(action);
+    }
+
+    /**
+     * initialize method
+     */
+    @Override
+    public void init() {
+
+        addParameter("name", name);
+        addParameter("context", context);
+        addParameter("endpoint_config", endpoint.toString());
+        addParameter("provider", getProvider());
+        addParameter("visibility", getVisibility());
+        addParameter("version", getVersion());
+        addParameter("description", getDescription());
+        addParameter("endpointType", getEndpointType());
+        if (getEndpointType().equals("secured")) {
+            addParameter("epUsername", epUsername);
+            addParameter("epPassword", epPassword);
+        }
+        addParameter("http_checked", getHttp_checked());
+        addParameter("https_checked", getHttps_checked());
+        addParameter("tags", getTags());
+        addParameter("tier", getTier());
+        addParameter("thumbUrl", getThumbUrl());
+        addParameter("tiersCollection", getTiersCollection());
+
+
+        if (resourceBeanList.size() < 2) {
```

Comment:
add a comment to explain the logic
comment : APIM is designed to send 0 as resource count if there is 0 and 1 resources. From 2 resources it sends the correct count.